synapse/storage/databases/main/transactions.py

```diff
@@ -98,26 +97,27 @@ def _get_received_txn_response(self, txn, transaction_id, origin):
         else:
             return None
 
-    def set_received_txn_response(self, transaction_id, origin, code, response_dict):
-        """Persist the response we returened for an incoming transaction, and
+    async def set_received_txn_response(
+        self, transaction_id: str, origin: str, code: int, response_dict: JsonDict
+    ) -> None:
+        """Persist the response we returned for an incoming transaction, and
         should return for subsequent transactions with the same transaction_id
         and origin.
 
         Args:
-            txn
-            transaction_id (str)
-            origin (str)
-            code (int)
-            response_json (str)
+            transaction_id: The incoming transaction ID.
+            origin: The origin server.
+            code: The response code.
+            response_dict: The response, to be encoded into JSON.
         """
 
-        return self.db_pool.simple_insert(
+        await self.db_pool.simple_insert(
             table="received_transactions",
             values={
                 "transaction_id": transaction_id,
                 "origin": origin,
                 "response_code": code,
-                "response_json": db_binary_type(encode_canonical_json(response_dict)),
+                "response_json": memoryview(encode_canonical_json(response_dict)),
```

Comment:
This is done in a couple other places around the codebase. Rather than change one, perhaps we could change them all separately (and also stick in a comment explaining what `memoryview` does as I imagine that's why it was renamed in the first place?).

Reply:
I'll revert this change for now -- we can deal with it separately. (I moved it since there was only a single use.)